dataset/dataset.cpp

```diff
@@ -3,10 +3,12 @@
 /// @file
 /// @author Simon Heybrock
 #include "scipp/dataset/dataset.h"
+#include "scipp/common/index.h"
 #include "scipp/dataset/except.h"
 #include "scipp/dataset/unaligned.h"
 
 #include "dataset_operations_common.h"
+#include <complex>
```

Comment:
Why?

Reply:
Sigh, my IDE "fixed" up this from `cmath` to `complex` as it was more specific